src/build_queue.rs

```diff
@@ -188,6 +188,23 @@ impl BuildQueue {
     }
 }
 
+fn retry<T>(mut f: impl FnMut() -> Result<T>) -> Result<T> {
+    const MAX_ATTEMPTS: u8 = 3;
+    for attempt in 1.. {
+        match f() {
+            Ok(result) => return Ok(result),
+            Err(err) => {
+                if attempt > MAX_ATTEMPTS {
+                    return Err(err);
+                } else {
+                    log::warn!("got error {:?} on attempt {}, will try again", err, attempt);
```

Comment:
The error print is likely to be multiline, so it would be more readable at the end (and maybe on a newline after the warning).

Reply:
I updated the message